metricbeat/module/mysql/query/query.go

```diff
@@ -89,40 +89,47 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
 	return nil
 }
 
-func (m *MetricSet) fetchQuery(ctx context.Context, q query, reporter mb.ReporterV2) error {
-	if m.Config.ResponseFormat == "table" {
-		mss, err := m.db.FetchTableMode(ctx, q.Query)
+func (m *MetricSet) fetchQuery(ctx context.Context, query query, reporter mb.ReporterV2) error {
+	if query.ResponseFormat == "table" {
+		mss, err := m.db.FetchTableMode(ctx, query.Query)
 		if err != nil {
 			return err
 		}
 
 		for _, ms := range mss {
-			event := mb.Event{ModuleFields: common.MapStr{m.Config.Namespace: common.MapStr{}}}
-			if q.QueryNamespace != "" {
-				event.ModuleFields[m.Config.Namespace] = common.MapStr{q.QueryNamespace: sql.ToDotKeys(ms)}
-			} else {
-				event.ModuleFields[m.Config.Namespace] = sql.ToDotKeys(ms)
-			}
+			event := m.transformMapStrToEvent(query,ms)
 			reporter.Event(event)
 		}
 	} else {
-		ms, err := m.db.FetchVariableMode(ctx, q.Query)
+		ms, err := m.db.FetchVariableMode(ctx, query.Query)
 		if err != nil {
 			return err
 		}
 
-		event := mb.Event{ModuleFields: common.MapStr{m.Config.Namespace: common.MapStr{}}}
-		if q.QueryNamespace != "" {
-			event.ModuleFields[m.Config.Namespace] = common.MapStr{q.QueryNamespace: sql.ToDotKeys(ms)}
-		} else {
-			event.ModuleFields[m.Config.Namespace] = sql.ToDotKeys(ms)
-		}
+		event := m.transformMapStrToEvent(query,ms)
 		reporter.Event(event)
 	}
 
 	return nil
 }
 
+func (m *MetricSet) transformMapStrToEvent(query query, ms common.MapStr) mb.Event {
+	event := mb.Event{ModuleFields: common.MapStr{m.Config.Namespace: common.MapStr{}}}
+
+	data := ms
+	if query.DeDotEnabled {
```

Comment:
Would it be the oposite? When `!query.DeDotEnabled`, then we add the dots.

Reply:
It was finally replaced with `ReplaceUnderscores`